### PR TITLE
Add PeerConnection state machine

### PR DESCRIPTION
**Pull request by mickel8**

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -45,16 +45,16 @@ defmodule ExWebRTC.MixProject do
...
-      {:ex_dtls, "~> 0.14"},
-      {:ex_libsrtp, "~> 0.6"},
-      {:ex_rtp, "~> 0.2"},
-      {:ex_rtcp, "~> 0.1"},
+      {:ex_sdp, "~> 0.13.0"},
+      {:ex_ice, "~> 0.2.0"},
+      {:ex_dtls, "~> 0.14.0"},
+      {:ex_libsrtp, "~> 0.6.0"},
+      {:ex_rtp, "~> 0.2.0"},
+      {:ex_rtcp, "~> 0.1.0"},
```
Comment (LVala) [inline mix.exs:53]: Why add the `path` digit to versions? I hope you don't plan to introduce breaking changes in the libraries with `patch` updates 🙂

**Review by LVala — APPROVED** [review]

File: mix.exs (hunk 1 above)
Comment (mickel8) [reply mix.exs:53]: in packages that are in `0.x.x.`, breaking changes can be introduced in minor

File: lib/ex_webrtc/peer_connection.ex [hunk 2]
```diff
@@ -514,6 +545,32 @@ defmodule ExWebRTC.PeerConnection do
...
+  # TODO support :disconnected state - our ICE doesn't provide disconnected state for now
+  # TODO support :closed state
+  # the order of these clauses is important
+  defp next_conn_state(ice_state, dtls_state)
+
+  defp next_conn_state(ice_state, dtls_state) when ice_state == :failed or dtls_state == :failed,
+    do: :failed
+
+  defp next_conn_state(ice_state, dtls_state) when ice_state == :new and dtls_state == :new,
+    do: :new
```
[the remark this answers is not in the document]
Comment (LVala) [reply lib/ex_webrtc/peer_connection.ex:557]: Ah, right! Although, I think I still prefer the version with pattern matching.

[the remark this answers is not in the document]
Comment (mickel8) [reply lib/ex_webrtc/peer_connection.ex]: Fixed

**Pull request merged**